language/tools/move-cli/src/utils/credential.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) The Diem Core Contributors
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{bail, Context, Result};
+use move_command_line_common::env::MOVE_HOME;
+use std::{env, fs};
+use toml_edit::easy::Value;
+
+pub const MOVEY_API_KEY_PATH: &str = "/movey_api_key.toml";
+
+#[derive(Clone)]
+pub struct TestMode {
+    pub test_path: String,
+}
+
+pub fn get_move_home_path(test_mode: Option<TestMode>) -> String {
+    let mut move_home;
+    if let Some(test_mode) = test_mode {
+        move_home = env::var("TEST_MOVE_HOME").unwrap();
+        if !test_mode.test_path.is_empty() {
+            move_home.push_str(&test_mode.test_path);
+        }
+    } else {
+        move_home = MOVE_HOME.clone();
+    }
+    move_home
+}
+
+pub fn get_credential_path(test_mode: Option<TestMode>) -> String {
```

Comment:
The "are we in a test" logic is currently split between `credential.rs` and `movey_upload.rs`, which is somewhat messy. I think it would be much cleaner to put all of the test handling in a single code path in `movey_upload` and keep these API's simple.

My suggestion: 
- branch on `config.test_mode` in line 85 of `movey_upload` and call out to a separate `handle_test_upload()` function with all of the testing-related logic (including everything test-related in `credential.rs`)
- keep the main code path where `!config.test_mode` holds completely independent of test-related logic

I think this is probably worthwhile even if it forces you to do some code duplication in `movey_upload`--I just want to make sure that folks touching this code in the future have a very clear idea of what logic is for testing and what is for production.

Reply:
We've split the test from the main logic, but it may be difference from what you imagined. Please help us check again.